otelcol/config.go

```diff
@@ -153,5 +154,41 @@ func (cfg *Config) Validate() error {
 			return fmt.Errorf("service::pipelines::%s: references exporter %q which is not configured", pipelineID, ref)
 		}
 	}
+
+	// Check that all pipelines reference only configured components.
+	for pipelineID, pipeline := range cfg.Service.PipelinesWithPipelineID {
```

Comment:
do we need a second for loop here?

Reply:
This is a duplicate of the `for pipelineID, pipeline := range cfg.Service.Pipelines {` loop above it.

I didn't want to handle the merging of `cfg.Service.Pipelines` into `cfg.Service.PipelinesWithPipelineID` via the Validate function (it is instead handled in `service.initGraph`) so I instead opted to validate both `cfg.Service.Pipelines` and `cfg.Service.PipelinesWithPipelineID`.

The line above, `if err := cfg.Service.Validate(); err != nil`, will validate that only one of the 2 is set, so in reality only 1 for the 2 for loops is ever entered.